python/lsst/summit/utils/guiders/reading.py

```diff
@@ -0,0 +1,724 @@
+
+        Returns
+        -------
+        axisMap : `dict[str, int]`
+            Dictionary with detector names as keys and axis mapping as values.
+            The axis mapping is 1 for X and 0 for Y.
+        """
+        axisMap: dict[str, int] = {}
+        for detName in self.guiderNameMap.keys():
+            detNum = self.guiderNameMap[detName]
+            detector = self.camera[detNum]
+            nq = detector.getOrientation().getNQuarter()
+            axisMap[detName] = 0 if nq % 2 == 0 else 1
+        return axisMap
+
+    def getRoiAmpNames(self, dataset: dict[str, Stamps]) -> dict[str, str]:
+        """Get the name of the amplifier used for the ROI for each guider
+        detector in the focal plane.
+
+        Returns
+        -------
+        ampNames : dict[str, str]
+            Dictionary with detector names as keys and amplifier names as
+            values.
+        """
+        ampNames: dict[str, str] = {}
+        for detName in dataset.keys():
+            md = dataset[detName].metadata.toDict()
+            # also get the ampName
+            segment = md["ROISEG"]
+            ampName = "C" + segment[7:]
+            ampNames[detName] = ampName
+        return ampNames
+
+    def getHeaderInfo(
+        self, raw: Stamps, nstamps: int, freq: float, cam_rot_angle: float = 0.0
+    ) -> dict[str, str | float]:
+        info: dict[str, str | float] = {}
+        m = raw.metadata.toDict()
+        info["n_stamps"] = nstamps
+        info["start_time"] = m["GDSSTART"]
+        info["roi_cols"] = m["ROICOLS"]
+        info["roi_rows"] = m["ROIROWS"]
+        info["FREQ"] = float(freq)
+        info["expid"] = int(m["DAYOBS"]) * 100000 + int(m["SEQNUM"])
+        info["filter"] = m["FILTBAND"]
+        info["SHUTTIME"] = float(m.get("SHUTTIME", 30.0))
+        info["AZSTART"] = float(m.get("AZSTART", 0))
+        info["ELSTART"] = float(m.get("ELSTART", 0))
+        info["AZEND"] = m.get("AZEND", None)
+        info["ELEND"] = m.get("ELEND", None)
+        info["SEEING"] = m.get("SEEING", None)
+        info["CAM_ROT_ANGLE"] = cam_rot_angle
+        return info
+
+    def printHeaderInfo(self, header: dict[str, str | float]) -> None:
+        # TODO: reinstate dataId and filter if necessary
+        print(f"Data Id: {header['expid']}, filter-band: {header['filter']}")
+        print(f"ROI Shape (row, col): {header['roi_rows']}, {header['roi_cols']}")
+        mystr = f"With nstamps {int(header['n_stamps'])}"
+        mystr = mystr + f" at {header['FREQ']} Hz"
+        print(mystr)
+        print(f"Acq. Start Time: {header['start_time']}")
+
+    @property
+    def guiderNames(self):
+        """Get the names of the guider detectors."""
+        return list(self.guiderNameMap.keys())
+
+    @property
+    def guiderIds(self):
+        """Get the ids of the guider detectors."""
+        return list(self.guiderNameMap.values())
+
+
+# TODO: Check missing stamps
+def getGuiderStamps(
```

Comment:
This function does two tasks:
1) rename the guiders if the obs < 20250509
2) convert the `guider_raw` object to a given view (dvcs)

To make things simpler, I made two new functions:
1) `applyGuiderNameSwapIfNeeded` -> perfoms task 1 
2) `convertRawStampsToView` -> performs task 2